functorch/_src/aot_autograd.py

```diff
@@ -1133,54 +1260,58 @@ def remove_dupe_args(args):
     def add_dupe_args(args):
         return [args[add_dupe_map[i]] for i in range(duped_arg_len)]
 
-    def maybe_wrap_debug(f):
-        if not config.debug_assert:
-            return f
-
-        @wraps(f)
-        def debug_wrapper(*args):
-            # Test that the computed remove/add arg functions are an inverse
-            new_args = add_dupe_args(remove_dupe_args(args))
-            seen = {}
-            for i, (x, y) in enumerate(zip(new_args, args)):
-                seen[y] = None
-                assert x is y, format_guard_bug_msg(
-                    aot_config,
-                    f"{describe_input(i, aot_config)} would be a duplicate of "
-                    f"{describe_input(add_dupe_map[i], aot_config)}"
-                )
-            # This is only an error if there is metadata mutation on both of
-            # the duped arguments; in this case, we need to know what order
-            # the metadata mutation applies in.  You'll get the correct result
-            # otherwise, because a graph that assumes distinct inputs works if
-            # you dupe the inputs (the gradient contributions from each input
-            # will get summed up appropriately.)
-            """
-            assert len(seen) == unique_args, format_guard_bug_msg(aot_config,
-                f"there would be {unique_args} distinct arguments"
-            )
-            """
-            return f(*args)
-
-        return debug_wrapper
-
-    # Fastpath
-    if not dropped_args:
-        return maybe_wrap_debug(aot_dispatch_deduplicated_autograd(flat_fn, flat_args, aot_config))
-
     deduped_flat_args = remove_dupe_args(flat_args)
 
     @wraps(flat_fn)
     def wrapped_flat_fn(*args):
         return flat_fn(*add_dupe_args(args))
 
-    compiled_fn = aot_dispatch_deduplicated_autograd(wrapped_flat_fn, deduped_flat_args, aot_config)
+    compiled_fn = compiler_fn(wrapped_flat_fn, deduped_flat_args, aot_config)
+
+    if not hasattr(compiled_fn, "_boxed_call"):
+        compiled_fn = make_boxed_func(compiled_fn)
 
     @wraps(compiled_fn)
-    def wrapped_compiled_fn(*args):
-        return compiled_fn(*remove_dupe_args(args))
+    def wrapped_compiled_fn(args):
+        deduped_args = remove_dupe_args(args)
+        args.clear()
+        return compiled_fn(deduped_args)
+    wrapped_compiled_fn._boxed_call = True
+
+    # This can be uncommented when we properly guard for duplicates,
```

Comment:
Just to spell this out - is the idea that we **know** that this is a problem today, so we definitely want the loud error to happen by default. But once we add the guarding for duplicates in dynamo, then we'll be more confident that dupe args work default, and so we'll be comfortable with the debug checks not running by default?

Reply:
yup